src/metatensor/metatensor.cpp

```diff
@@ -0,0 +1,958 @@
+the atomic type of the atoms that have been specified using the `SPECIES1` flag,
+the second number is the atomic number of the atoms that have been specified
+using the `SPECIES2` flag and so on.
+
+`METATENSOR` action also accepts the following options:
+
+- `EXTENSIONS_DIRECTORY` should be the path to a directory containing
+  TorchScript extensions (as shared libraries) that are required to load and
+  execute the model. This matches the `collect_extensions` argument to
+  `MetatensorAtomisticModel.export` in Python.
+- `NO_CONSISTENCY_CHECK` can be used to disable internal consistency checks;
+- `SELECTED_ATOMS` can be used to signal the metatensor models that it should
+  only run its calculation for the selected subset of atoms. The model still
+  need to know about all the atoms in the system (through the `SPECIES`
+  keyword); but this can be used to reduce the calculation cost. Note that the
+  indices of the selected atoms should start at 1 in the PLUMED input file, but
+  they will be translated to start at 0 when given to the model (i.e. in
+  Python/TorchScript, the `forward` method will receive a `selected_atoms` which
+  starts at 0)
+
+Here is another example with all the possible keywords:
+
+\plumedfile
+soap: METATENSOR ...
+    MODEL=soap.pt
+    EXTENSION_DIRECTORY=extensions
+    NO_CONSISTENCY_CHECK
+
+    SPECIES1=1-10
+    SPECIES2=11-20
+    SPECIES_TO_TYPES=8,13
+
+    # only run the calculation for the Aluminium (type 13) atoms, but
+    # include the Oxygen (type 8) as potential neighbors.
+    SELECTED_ATOMS=11-20
+...
+\endplumedfile
+
+\par Collective variables and metatensor models
+
+Collective variables are not yet part of the [known outputs][mts_outputs] for
+metatensor models. Until the output format is standardized, this action expects
+the following:
+
+- the output name should be `"plumed::cv"`;
+- the output should contain a single [block][mts_block];
+- the output samples should be named `["system", "atom"]` for per-atom outputs;
+  or `["system"]` for global outputs. The `"system"` index should always be 0,
+  and the `"atom"` index should be the index of the atom (between 0 and the
+  total number of atoms);
+- the output should not have any components;
+- the output can have arbitrary properties;
+- the output should not have any explicit gradients, all gradient calculations
+  are done using autograd.
+
+*/ /*
+
+[TorchScript]: https://pytorch.org/docs/stable/jit.html
+[mts_models]: https://docs.metatensor.org/latest/atomistic/index.html
+[mts_tutorials]: https://docs.metatensor.org/latest/examples/atomistic/index.html
+[mts_outputs]: https://docs.metatensor.org/latest/atomistic/outputs.html
+[mts_block]: https://docs.metatensor.org/latest/torch/reference/block.html
+
+*/
+//+ENDPLUMEDOC
+
+/*INDENT-OFF*/
+#if !defined(__PLUMED_HAS_LIBTORCH) || !defined(__PLUMED_HAS_METATENSOR)
+
+namespace PLMD { namespace metatensor {
```

Comment:
@Luthaf I think that you should place the namespace on a separate line for our script to detect it.

Notice that you can run very quickly the check on your computer with `cd src ; make plumedcheck`. It just needs gawk to be installed

Reply:
sure!